engine/script/src/script_ddf.cpp

```diff
@@ -320,8 +321,10 @@ namespace dmScript
                     if (!nil_val)
                     {
                         const dmDDF::Descriptor* d = f->m_MessageDescriptor;
-                        bool is_vector3 = strncmp(d->m_Name, DDF_TYPE_NAME_VECTOR3, sizeof(DDF_TYPE_NAME_VECTOR3)) == 0;
-                        bool is_point3 = strncmp(d->m_Name, DDF_TYPE_NAME_POINT3, sizeof(DDF_TYPE_NAME_POINT3)) == 0;
+                        const dmhash_t name_hash = d->m_NameHash;
+                        bool is_vector3 = name_hash == DDF_TYPE_NAME_HASH_VECTOR3;
+                        bool is_point3 = name_hash == DDF_TYPE_NAME_HASH_POINT3;
```

Comment:
finally switching to hash comparisons. not sure I could properly measure the gain. But we know it's better at least :)